src/main/java/com/atlauncher/gui/dialogs/instancesettings/GeneralInstanceSettingsTab.java

```diff
@@ -123,12 +125,38 @@ private void setupComponents() {
         }
 
         add(enableDiscordIntegration, gbc);
+
+        // Join server on launch
+
+        gbc.gridx = 0;
+        gbc.gridy++;
+        gbc.insets = UIConstants.LABEL_INSETS;
+        gbc.anchor = GridBagConstraints.BASELINE_TRAILING;
+
+        JLabelWithHover initialJoinServerAddressLabel = new JLabelWithHover(
+            GetText.tr("Join Server On Launch") + ":", HELP_ICON,
+            GetText.tr(
+                "Enter the server address if you want to join a Minecraft server when you launch the game, " +
+                    "leave it empty if you don't want to join a server after launching the game."
+            )
+        );
+
+        add(initialJoinServerAddressLabel, gbc);
+
+        gbc.gridx++;
+        gbc.insets = UIConstants.FIELD_INSETS;
+        gbc.anchor = GridBagConstraints.BASELINE_LEADING;
+        initialJoinServerAddress = new JTextField(13);
+        initialJoinServerAddress.setText(instance.launcher.initialJoinServerAddress);
+
+        add(initialJoinServerAddress, gbc);
     }
 
     public void saveSettings() {
         this.instance.launcher.account = ((ComboItem<String>) account.getSelectedItem()).getValue();
         this.instance.launcher.enableDiscordIntegration = ((ComboItem<Boolean>) enableDiscordIntegration
                 .getSelectedItem()).getValue();
+        this.instance.launcher.initialJoinServerAddress = initialJoinServerAddress.getText();
```

Comment:
We'll want to do some validation here to make sure it's valid, otherwise any old garbage could be entered here.

I guess something simple like check that there's either no `:` or if there is a `:` that there are only numbers after the `:` between 1 and 65535. That should be simple enough without having to get into doing 500 character long regexes.

Reply:
> We'll want to do some validation here to make sure it's valid, otherwise any old garbage could be entered here.
> 
> I guess something simple like check that there's either no `:` or if there is a `:` that there are only numbers after the `:` between 1 and 65535. That should be simple enough without having to get into doing 500 character long regexes.

I wanted to discuss about that, but I didn't find any section for the suggestions and there is no discussions second in this repository and I don't plan on mixing the issues with the questions for organization, and an example of how ATLauncher handles the input validation so I do something similar here for consistency

Reply:
> I guess something simple like check that there's either no `:` or if there is a `:` that there are only numbers after the `:` between 1 and 65535. That should be simple enough without having to get into doing 500 character long regexes.

Should the Patterns or Input text validators go to the source code? I want to add some tests for validating input and output, regarding of using Pattern or validating the input using java internal functions like (contains, isEmpty etc..) it could be a function and not directly a Pattern because we might change the Pattern later and have to change it from different places, also tests are helpful to make sure we are safe

Reply:
> Would make sense to have a utility class for validation, like ValidationUtils for instance (or simply just Validation)
Where should this File go? Maybe (src/main/java/com/atlauncher/utils)?? I'm not sure if there is a better package

Reply:
I'm getting build failure exception:

```console

* What went wrong:
Execution failed for task ':checkLicenseMain'.
> License violations were found: src/main/java/com/atlauncher/utils/ValidationUtils.java

```

it's a simple class that doesn't use any imports other than `java.util.regex.Pattern`

Reply:
> We'll want to do some validation here to make sure it's valid

Done, there are common logic in the validation and the error dialogs, maybe we should extract them later